modules/git/hook.go

```diff
@@ -12,46 +11,49 @@ import (
 	"strings"
 
 	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
 	"code.gitea.io/gitea/modules/util"
 )
 
 // hookNames is a list of Git server hooks' name that are supported.
-var hookNames = []string{
-	"pre-receive",
-	"update",
-	"post-receive",
+func GetHookNames() map[string]string {
+	return map[string]string{
+		"pre-receive":  setting.GitHookPrereceiveName,
+		"update":       setting.GitHookUpdateName,
+		"post-receive": setting.GitHookPostreceiveName,
+	}
+}
+
+// Hook represents a Git hook.
+type Hook struct {
+	name     string
+	IsActive bool   // Indicates whether repository has this hook.
+	Content  string // Content of hook if it's active.
+	Sample   string // Sample content from Git.
+	path     string // Hook file path.
 }
 
 // ErrNotValidHook error when a git hook is not valid
 var ErrNotValidHook = errors.New("not a valid Git hook")
 
 // IsValidHookName returns true if given name is a valid Git hook.
 func IsValidHookName(name string) bool {
-	for _, hn := range hookNames {
+	for hn := range GetHookNames() {
 		if hn == name {
 			return true
 		}
 	}
 	return false
 }
 
-// Hook represents a Git hook.
-type Hook struct {
-	name     string
-	IsActive bool   // Indicates whether repository has this hook.
-	Content  string // Content of hook if it's active.
-	Sample   string // Sample content from Git.
-	path     string // Hook file path.
-}
-
 // GetHook returns a Git hook by given name and repository.
 func GetHook(repoPath, name string) (*Hook, error) {
 	if !IsValidHookName(name) {
 		return nil, ErrNotValidHook
 	}
 	h := &Hook{
 		name: name,
-		path: path.Join(repoPath, "hooks", name+".d", name),
+		path: path.Join(repoPath, "hooks", name+".d", GetHookNames()[name]),
```

Comment:
Interesting. It is possible to pass directories as well via the new config options.
Is that intended behavior?
I'm not against that, but want to know if we want to allow setting relative directories in that method.

Reply:
Oh, also, isn't that the wrong path?
Shouldn't it be `filepath.Join`?

Reply:
> Interesting. It is possible to pass directories as well via the new config options. Is that intended behavior? I'm not against that, but want to know if we want to allow setting relative directories in that method.

Thanks for bringing this up. I've just implemented a validation to ensure that only filenames, and not paths, can be passed in through the new config options.

> Oh, also, isn't that the wrong path?
Shouldn't it be filepath.Join?

it is the wrong path, thanks for pointing it out